### PR TITLE
Migrate to connect-es

### PR DESCRIPTION
**Pull request by edaniels**
Depends on https://github.com/viamrobotics/viam-typescript-sdk/pull/368

There's nothing very interesting here and testing so far seems to have this work.

### CODE REVIEW

File: web/frontend/package.json [hunk 1]
```diff
@@ -59,8 +53,7 @@
     "@viamrobotics/prime": "0.5.9",
     "@viamrobotics/prime-blocks": "^0.1.3",
     "@viamrobotics/prime-core": "^0.0.87",
-    "@viamrobotics/rpc": "0.2.3",
-    "@viamrobotics/sdk": "0.25.0",
+    "@viamrobotics/sdk": "file:///Users/eric/viam/viam-typescript-sdk",
```
Comment (njooma) [inline web/frontend/package.json:56]: Obvi this will have to be reverted once the other PR is released

**Review by njooma — APPROVED** [review]
Seems like the changes are all of the same type. Everything reads a lot cleaner

File: web/frontend/src/lib/components/robot-client.svelte [hunk 2]
```diff
@@ -439,22 +450,20 @@ const start = () => {
...
-  const ae = authEntity ?? bakedAuth.authEntity;
-  if (c && ae) {
+  if (c) {
     sdkCreds = {
       type: c.type as CredentialType,
       payload: c.payload,
-      authEntity: ae,
-    };
+      authEntity: c.authEntity ?? bakedAuth.authEntity,
+    } as Credential;
```
Comment (zaporter-work) [inline web/frontend/src/lib/components/robot-client.svelte:463]: [nit] use `satisfies Credential` here or add more type info to `sdkCreds`

File: web/frontend/src/api/movement-sensor.ts [hunk 3]
```diff
@@ -1,193 +1,61 @@
...
-          } else {
-            resolve(res);
-          }
-        });
-      }
-    );
-
-  return response?.toObject().orientation;
+  const client = new MovementSensorClient(robotClient, name, {
+    requestLogger: rcLogConditionally,
```
Comment (zaporter-work) [inline web/frontend/src/api/movement-sensor.ts:13]: [aside] Do you know if these clients expensive to construct? I traced the trail down through connect-es PromiseClient but didn't go much further.

**Review by zaporter-work — APPROVED** [review]
Did not manually test due to hardcoded file in package.json.

Code LGTM

File: web/frontend/src/lib/components/robot-client.svelte (hunk 2 above)
Comment (edaniels) [reply web/frontend/src/lib/components/robot-client.svelte:463]: tyty

File: web/frontend/src/components/do-command/get-client-by-type.ts [hunk 4]
```diff
@@ -1,44 +1,57 @@
...
-  gantry: 'gantryService',
-  generic: 'genericService',
-  gripper: 'gripperService',
-  input_controller: 'inputControllerService',
-  motion: 'motionService',
-  motor: 'motorService',
-  movement_sensor: 'movementSensorService',
-  navigation: 'navigationService',
-  power_sensor: 'powerSensorService',
-  sensors: 'sensorsService',
```
Comment (ethanlook) [inline web/frontend/src/components/do-command/get-client-by-type.ts:21]: Does the sensors service no longer exist? I was under the impression it still existed.

File: web/frontend/src/api/sensors.ts [hunk 5]
```diff
@@ -1,20 +1,7 @@
...
-        } else {
-          resolve(res);
-        }
-      });
-    }
-  );
-
-  return response?.toObject().sensorNamesList ?? [];
+  const request = new sensorsApi.GetSensorsRequest({ name });
+  const resp = await robotClient.sensorsService.getSensors(request);
```
Comment (ethanlook) [inline web/frontend/src/api/sensors.ts:5]: Why not create `SensorsClient` like the other wrappers?

File: web/frontend/src/api/power-sensor.ts
Comment (ethanlook) [inline web/frontend/src/api/power-sensor.ts:1]: Why are the responses for `getVoltage`, `getCurrent`, and `getPower` arrays now?

File: web/frontend/src/lib/components/robot-client.svelte [hunk 6]
```diff
@@ -196,71 +201,75 @@ const fetchCurrentSessions = async () => {
...
+      $statusStream = null;
     }
-    $statusStream = null;
-  });
-  $statusStream.on('end', () => {
-    // eslint-disable-next-line no-console
-    console.error('done streaming robot status');
-    $statusStream = null;
-  });
+  })().catch(console.error);
```
Comment (ethanlook) [inline web/frontend/src/lib/components/robot-client.svelte:272]: Will this catch actually catch anything since the whole thing is wrapped in a try/catch?

**Review by ethanlook — COMMENTED** [review]
Code LGTM, manual testing now

File: web/frontend/src/components/do-command/get-client-by-type.ts (hunk 4 above)
Comment (edaniels) [reply web/frontend/src/components/do-command/get-client-by-type.ts:21]: there's no reified client for it other than the promise client. is anyone running a do command against a sensors service though?

File: web/frontend/src/api/sensors.ts (hunk 5 above)
Comment (edaniels) [reply web/frontend/src/api/sensors.ts:5]: because the TS SDK never had one

Comment (edaniels) [reply web/frontend/src/api/sensors.ts:5]: note there's a SensorClient but not a SensorsClient

Comment (ethanlook) [reply web/frontend/src/api/sensors.ts:5]: not a problem then!

File: web/frontend/src/api/power-sensor.ts
Comment (edaniels) [reply web/frontend/src/api/power-sensor.ts:1]: they previously were in the TS SDK https://github.com/viamrobotics/viam-typescript-sdk/blob/9cb36cb4b0eb407996fb7b14179085c7e2f883d9/src/components/power-sensor/power-sensor.ts

File: web/frontend/src/lib/components/robot-client.svelte (hunk 6 above)
Comment (edaniels) [reply web/frontend/src/lib/components/robot-client.svelte]: nope good... catch

File: web/frontend/src/api/power-sensor.ts
Comment (ethanlook) [reply web/frontend/src/api/power-sensor.ts:1]: Ah interesting, what is the bool that is the second entry? Could we use array destructuring?

Comment (edaniels) [reply web/frontend/src/api/power-sensor.ts:1]: I think it's something about it being AC or not

File: web/frontend/src/components/do-command/get-client-by-type.ts (hunk 4 above)
Comment (edaniels) [reply web/frontend/src/components/do-command/get-client-by-type.ts:21]: if a problem comes up with it, it wont be hard to get it working again!

**Comment by ethanlook** [comment]
Manual testing LGTM - I wasn't seeing any operations but not sure that I would with the fakes. There are hopefully very few users on this so I think we're safe and can fix forward if anything comes up.

**Comment by edaniels** [comment]
> Manual testing LGTM - I wasn't seeing any operations but not sure that I would with the fakes. There are hopefully very few users on this so I think we're safe and can fix forward if anything comes up.

Ah yeah I managed to get one using a motor gofor, many of them are too instantaneous though

**Pull request merged**